CHANGELOG.md

```diff
@@ -21,10 +44,89 @@ All notable changes to DSCv3 are documented in this file. The format is based on
 This section includes a summary of user-facing changes since the last release. For the full list of
 changes since the last release, see the [diff on GitHub][unreleased].
 
-<!-- Unreleased comparison link -->
+<!-- Unreleased comparison link - always update version to match last release tag-->
 [unreleased]: https://github.com/PowerShell/DSC/compare/v3.0.0-alpha.5...main
 
-<!-- Add entries between releases under the appropriate section heading here  -->
+<!--
+    Unreleased change entry instructions:
+
+    Add entries between releases under the appropriate section heading here. When you need to add
+    a change, make sure it's under one of the H3s, not this H2. Use one of the following snippets
+    to add the change entry:
+
+    - docs-changelog-entry-single-issue-pr: Adds a new changelog list entry with one related issue
+      and PR. Use this when a change entry only has one related issue and/or pull request.
+
+    - docs-changelog-entry-multi-issue-pr: Adds a new changelog list entry with sub-lists for
+      issues and PRs. Use this when a change entry has more than one related issue and/or pull
+      request.
+
+    When you're ready to update the unreleased changelog entries for a new release, use the
+    docs-changelog-release-heading snippet to create the new release heading after this comment and
+    before the first H3 for the changes.
+
+    After doing so, rename the unreleased reference links from `ur.##` to `<prefix>.##`, where
+    <prefix> is a two-character prefix for the release. For alpha releases, we use `a#`, like `a5`
+    for the `v3.0.0.0-alpha.5` release. Leave the release links under the release section.
+-->
+
+### Changed
+
+- The [envvar()][a5.12] function now raises an error when the specified environment variable
+  doesn't exist instead of returning an empty string. This change helps reduce unexpected and
+  difficult to diagnose errors when a configuration expects a value from the environment variable.
+
+  <details><summary>Related work items</summary>
+
+  - Issues: [#336][#336]
+  - PRs: [#358][#358]
+
+  <details>
+
+### Added
+
+- Added configuration functions:
+
+  - New mathematics functions include [add()][ur.01], [div()][ur.02], [max()][ur.03],
+    [min()][ur.04], [mod()][ur.05], [mul()][ur.06], and [sub()][ur.07]. The mathematics functions
+    only operate on integer values.
+
+  - The [reference()][ur.08] function enables you to reference the result output for other
```

Comment:
Might be worth calling out that you can only reference a resource that has already executed previously, so order is important.  You'll get an error if you try to reference a resource that hasn't executed yet or doesn't exist in the configuration.

Reply:
Updated to clarify recommendation for adding the referenced instance to the `dependsOn` property when using `reference()`.